src/main/scala/viper/server/frontends/lsp/Coordinator.scala

```diff
@@ -0,0 +1,80 @@
+/**
+  * This Source Code Form is subject to the terms of the Mozilla Public
+  * License, v. 2.0. If a copy of the MPL was not distributed with this
+  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+  *
+  * Copyright (c) 2011-2020 ETH Zurich.
+  */
+
+package viper.server.frontends.lsp
+
+import java.util.concurrent.CompletableFuture
+
+import org.eclipse.lsp4j.TextDocumentItem
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+object Coordinator {
```

Comment:
I don't remember reading about this concept in your thesis. Is it described somewhere?

Reply:
Yes, it is descibed in a sub-section of the Language server chapter. (1.3.3. in the draft version)

Reply:
Thanks!